pennylane/interfaces/execution.py

```diff
@@ -617,36 +714,15 @@ def gradient_fn(internal_tapes):
         # in this case would have ambiguous behaviour.
         raise ValueError("Gradient transforms cannot be used with grad_on_execution=True")
 
-    mapped_interface = INTERFACE_MAP[config.interface]
+    ml_boundary = _get_ml_framework_boundary(interface, _grad_on_execution)
     try:
-        if mapped_interface == "autograd":
-            from .autograd import execute as _execute
-
-        elif mapped_interface == "tf":
-            import tensorflow as tf
-
-            if not tf.executing_eagerly() or "autograph" in interface:
-                from .tensorflow_autograph import execute as _execute
-
-                _execute = partial(_execute, grad_on_execution=_grad_on_execution)
-
-            else:
-                from .tensorflow import execute as _execute
-
-        elif mapped_interface == "torch":
-            from .torch import execute as _execute
-
-        elif mapped_interface == "jax":
-            _execute = _get_jax_execute_fn(interface, tapes)
-
-        res = _execute(
+        res = ml_boundary(
             tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=max_diff
         )
-
     except ImportError as e:

```

Comment:
doesn't the ImportError happen in `_get_ml_framework_boundary` (above) now?

Reply:
I think the error is tested by mocking some functions in each library.

But the actual error would be raised on importing the relevant function.

Reply:
so can we remove this?

Reply:
Some tests would fail, and need to be removed. See: https://github.com/PennyLaneAI/pennylane/blob/6e8b491cab67f1c87de2a54b178c37653440069b/tests/interfaces/test_jax.py#L32

Reply:
is it too much to do in this PR?

Reply:
removing these import error tests. That's not where import errors are actually going to get raised.